AEFeinstein/Super-2024-Swadge-FW
# PNG Screenshots

### Description

This PR adds PNG screenshots using `stb_image_write`, and lets you take them with F12 instead of only with a replay script. To support that in a sane way, rawdraw is also updated for upstream changes that add defines for the key constants. And the emu extension interface is updated a bit to make working with modifier keys easier.

### Test Instructions

Run the emulator, hit F12 and take some screenshots. They should be written out to `screenshot-xxxx.png` with xxxx being a seconds+milliseconds unix timestamp. Run the emulator with `-r` and take some screenshots. Replay the recording with `-p`. The screenshot-taking should be recorded and replayed properly.

### Ticket Links



### Readiness Checklist
